### ISSUE TITLE
makeXlsxRowFromRow doesn't skip empty cells

### ISSUE THREAD

**Issue by nickylogan** [tealeg/xlsx#826]
Sheets with a huge MaxCol and a lot of empty cells seem to cause memory leak, as shown below—it keeps growing the cell slice. 

![image](https://github.com/user-attachments/assets/f6ef5d7d-36c5-41b2-910e-6632f3380ff4)

Checking the following function, it seems to not skip empty cells. Is this behavior intentional?
![image](https://github.com/user-attachments/assets/c796631c-87fe-46ca-ad69-707a45bf7001)

**Comment by tealeg** [comment]
Hey @nickylogan  - yup, there's a [specific option](https://pkg.go.dev/github.com/tealeg/xlsx/v3#SkipEmptyCells) you can pass into `ForEachCell` that will fix that for you:

```
err := row.ForEachCell(func(cell *Cell) error {
   ...
}, xlsx.SkipEmptyCells)
```

**Comment by nickylogan** [comment]
@tealeg I'm using version v3.3.5, but it seems this is not in the latest tag (v3.310) either.

Refer to the function here:
https://github.com/tealeg/xlsx/blob/v3.3.10/xmlWorksheet.go#L647

**Comment by benedictjohannes** [comment]
> @tealeg I'm using version v3.3.5, but it seems this is not in the latest tag (v3.3.10) either.
> 
> Refer to the function here: https://github.com/tealeg/xlsx/blob/v3.3.10/xmlWorksheet.go#L647

Apparently, the latest tagged version does not include `SkipEmptyCells`.

> Sorry got carried away. Misunderstood what you were saying. What version of the code are you looking at? The current code uses `SkipEmptyCells` explicitly.
> 
> https://github.com/tealeg/xlsx/blob/cbf45342baf42535dc6bcc3dd629f7412f4c4eaf/xmlWorksheet.go#L720

Can you profile again with the latest `master` branch to see whether this issue has been sorted out @nickylogan ?  `go get github.com/tealeg/xlsx/v3@347697717945`  (latest commit as of now)

**Comment by tealeg** [comment]
@nickylogan ah yes, you're right I haven't tagged a release with that in yet.

**Comment by tealeg** [comment]
@nickylogan .. easily resolved, welcome to version 3.3.11 :-D 

https://github.com/tealeg/xlsx/releases/tag/v3.3.11
